axelrod/player.py

```diff
@@ -161,9 +161,12 @@ def __eq__(self, other):
                     setattr(other, attribute,
                             itertools.cycle(original_other_value))
 
-                if not (all(next(generator) == next(other_generator)
-                        for _ in range(200))):
-                    return False
+                for _ in range(200):
+                    try:
+                        if not next(generator) == next(other_generator):
```

Comment:
Good call on `!=`.

I believe the error was thrown by an attribute that was not a cycle and ran out of elements. I didn't check through for which one that was though (this change seemed to make sense anyway to me).